aws/adapter.go

```diff
@@ -67,6 +67,7 @@ type manifest struct {
 	instance      *instanceDetails
 	subnets       []*subnetDetails
 	filters       []*ec2.Filter
+	asgFilters    map[string][]string
```

Comment:
What is the use case for a list of values here? Maybe I'm not 100% understanding how the EC2 filter API works?

Reply:
So... the ASG filters are... basically useless. You can only do a boolean OR for multiple filters, so we'd have to take a pass over the results to compare tags anyways.... so since we're already iterating over the set whole set of ASGs, I am just doing a comparison of the tags. (Which I munge from the EC2 filter expression)

Reply:
Here's a clipping from cluster-autoscaler where they *do* use a autoscaling.Filter{}, but then have to iterate the result anyways: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/auto_scaling.go#L130-L145

So there *might* be a small optimisation here, where we also do both, but then we'd have to create yet another filter representation to carry along.